django/contrib/postgres/fields/array.py

```diff
@@ -70,9 +72,11 @@ def db_type(self, connection):
         size = self.size or ''
         return '%s[%s]' % (self.base_field.db_type(connection), size)
 
-    def get_prep_value(self, value):
+    def get_db_prep_value(self, value, connection, prepared=False):
+        if isinstance(self.base_field, GenericIPAddressField):
```

Comment:
Seems like a generic hook could be useful here so we aren't handling field particularities in the field (i.e. how will custom fields handle this problem?)